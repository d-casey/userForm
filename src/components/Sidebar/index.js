import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import '../../css/sidebar.css'
/*
  The next step for this component is to add each stages content to config->stageConfig. That way updating one file can be easy to add new
  progress bar items and sidebar content.
*/
const Sidebar = ({ appStage }) => {
  const userFormContent = () => {
    return (
      <Fragment>
        <h3>Sign Up Form</h3>
        <p>Required fiels are marked with an *</p>
        <p>Passwords must be:</p>
        <ul>
          <li>At least 10 characters long</li>
          <li>Contain at least one number</li>
          <li>Contain at least one upper case letter</li>
          <li>Contain at least one lower case letter</li>
        </ul>
      </Fragment>
    )
  }

  const privacyFormContent = () => {
    return (
      <Fragment>
        <h3>Privacy</h3>
        <p>Please let us know how you would like us to contact you.</p>
      </Fragment>
    )
  }

  const done = () => {
    return (
      <Fragment>
        <h3>Done!</h3>
      </Fragment>
    )
  }

  return (
    <div className="sidebar">
      { appStage === 0 && userFormContent() }
      { appStage === 1 && privacyFormContent() }
      { appStage === 2 && done() }
    </div>
  )
}

Sidebar.propTypes = {
  appStage: PropTypes.number
}

export default Sidebar
