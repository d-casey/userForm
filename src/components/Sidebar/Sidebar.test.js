import React from 'react'
import { create } from 'react-test-renderer'
import Sidebar from '.'

test('renders correctly when empty', () => {
  const component = create(<Sidebar />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('renders the correct content for the initial stage', () => {
  const component = create(<Sidebar appStage={0} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('renders the correct content for the second stage', () => {
  const component = create(<Sidebar appStage={1} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('renders the correct content for the third stage', () => {
  const component = create(<Sidebar appStage={2} />)
  expect(component.toJSON()).toMatchSnapshot()
})
