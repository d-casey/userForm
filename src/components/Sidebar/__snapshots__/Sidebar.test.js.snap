// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when empty 1`] = `
<div
  className="sidebar"
/>
`;

exports[`renders the correct content for the initial stage 1`] = `
<div
  className="sidebar"
>
  <h3>
    Sign Up Form
  </h3>
  <p>
    Required fiels are marked with an *
  </p>
  <p>
    Passwords must be:
  </p>
  <ul>
    <li>
      At least 10 characters long
    </li>
    <li>
      Contain at least one number
    </li>
    <li>
      Contain at least one upper case letter
    </li>
    <li>
      Contain at least one lower case letter
    </li>
  </ul>
</div>
`;

exports[`renders the correct content for the second stage 1`] = `
<div
  className="sidebar"
>
  <h3>
    Privacy
  </h3>
  <p>
    Please let us know how you would like us to contact you.
  </p>
</div>
`;

exports[`renders the correct content for the third stage 1`] = `
<div
  className="sidebar"
>
  <h3>
    Done!
  </h3>
</div>
`;
